WallsAndHoles/tilemaptoolmanager.cpp

```diff
@@ -1,53 +1,55 @@
 #include "tilemaptoolmanager.h"
 
-inline AbstractTileMapTool *tool2TileMapTool(QSharedPointer<AbstractTool> tool)
+inline AbstractTileMapTool *tool2TileMapTool(AbstractTool *tool)
 {
-    return static_cast<AbstractTileMapTool *>(tool.data());
+    return static_cast<AbstractTileMapTool *>(tool);
```

Comment:
In this case, it is impossible to add a non-AbstractTileMapTool to a TileMapToolManager, so if this causes a segfault for anyone, it will be for someone who modified tilemaptoolmanager.cpp. It will also probably be very easy to find considering the "tool2TileMapTool" calls everywhere.